crates/astria-sequencer/src/transaction/fees.rs

```diff
@@ -0,0 +1,683 @@
+            to.expect("to address should be `Some`"),
+            fee_asset,
+            expected_deposit_fee,
+            fee_payment_map,
+            "BridgeLockAction".to_string(),
+            action_index,
+        );
+    } else {
+        warn_if_asset_type_not_allowed(&act.fee_asset, allowed_fee_assets);
+    }
+    Ok(())
+}
+
+fn bridge_unlock_update_fees(
+    payment_info: &mut PaymentInfo,
+    transfer_base_fee: u128,
+    add_to_payment_map: bool,
+    action_index: u32,
+) -> Result<()> {
+    let PaymentInfo {
+        from,
+        to,
+        allowed_fee_assets,
+        fee_asset,
+        fees_by_asset,
+        fee_payment_map,
+    } = payment_info;
+
+    fees_by_asset
+        .entry(fee_asset.to_ibc_prefixed())
+        .and_modify(|amt| *amt = amt.saturating_add(transfer_base_fee))
+        .or_insert(transfer_base_fee);
+    if add_to_payment_map {
+        err_if_asset_type_not_allowed(fee_asset, allowed_fee_assets)?;
+        increase_fees(
+            from.expect("from address should be `Some`"),
+            to.expect("to address should be `Some`"),
+            fee_asset,
+            transfer_base_fee,
+            fee_payment_map,
+            "BridgeUnlockAction".to_string(),
+            action_index,
+        );
+    } else {
+        warn_if_asset_type_not_allowed(fee_asset, allowed_fee_assets);
+    }
+    Ok(())
+}
+
+fn bridge_sudo_change_update_fees(
+    payment_info: &mut PaymentInfo,
+    bridge_sudo_change_base_fee: u128,
+    add_to_payment_map: bool,
+    action_index: u32,
+) -> Result<()> {
+    let PaymentInfo {
+        from,
+        to,
+        allowed_fee_assets,
+        fee_asset,
+        fees_by_asset,
+        fee_payment_map,
+    } = payment_info;
+
+    fees_by_asset
+        .entry(fee_asset.to_ibc_prefixed())
+        .and_modify(|amt| *amt = amt.saturating_add(bridge_sudo_change_base_fee))
+        .or_insert(bridge_sudo_change_base_fee);
+    if add_to_payment_map {
+        err_if_asset_type_not_allowed(fee_asset, allowed_fee_assets)?;
+        increase_fees(
+            from.expect("from address should be `Some`"),
+            to.expect("to address should be `Some`"),
+            fee_asset,
+            bridge_sudo_change_base_fee,
+            fee_payment_map,
+            "BridgeSudoChangeAction".to_string(),
+            action_index,
+        );
+    } else {
+        warn_if_asset_type_not_allowed(fee_asset, allowed_fee_assets);
+    }
+    Ok(())
+}
+
+fn handle_fee_asset_change(act: &FeeAssetChangeAction, allowed_fee_assets: &mut Vec<IbcPrefixed>) {
+    match act {
+        FeeAssetChangeAction::Addition(asset) => {
+            allowed_fee_assets.push(asset.clone().into());
+        }
+        FeeAssetChangeAction::Removal(asset) => {
+            allowed_fee_assets.retain(|a| *a != IbcPrefixed::from(asset));
+        }
+    }
+}
+
+fn handle_fee_change(act: &FeeChangeAction, current_fees: &mut Fees) {
+    match act.fee_change {
```

Comment:
i just realized this should also have a `BridgeUnlock` variant so the fees for `BridgeUnlock` can be updated, we should do this in a follow up